server/py/services/api/crud/model_monitoring/deployment.py

```diff
@@ -1824,9 +1824,17 @@ def set_credentials(
                 self.check_if_credentials_are_set()
                 if self._is_the_same_cred(stream_profile_name, tsdb_profile_name):
                     logger.debug(
-                        "The same credentials are already set for the project - aborting with no error",
+                        "The same credentials are already set for the project - ensuring TSDB tables exist",
                         project=self.project,
                     )
+                    # Even if credentials match, ensure TSDB tables exist (ML-11807).
+                    # This handles cases where tables were deleted or don't exist yet.
+                    # The create_tables() call is idempotent (uses CREATE TABLE IF NOT EXISTS).
```

Comment:
This depends on the connector though. Are they all idempotent? They don't all use `CREATE TABLE IF NOT EXISTS`, that's certain.

Reply:
fixed